### PR TITLE
CLOUDP-90359: Tested split-horizon and added docs

### PR DESCRIPTION
**Pull request by priyolahiri**
Tested split horizon, added docs and example manifests for deploying MongoDB with external access.

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you signed all of your commits?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Review by irajdeep — APPROVED** [review]
LGTM 👍

File: docs/external_access.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+### Connect to the MongoDB deployment from outside the Kubernetes cluster
+
+This is an example to connect to the MongoDB cluster with Mongo shell. Use the certificates from the previous step. Replace the values in the command from the preceeding steps.
+
+```sh
+mongosh --tls --tlsCAfile ca.crt --tlsCertificateKeyFile key.pem --username my-user --password <your-admin-password> mongodb://<domain-rs-1>:31181,<domain-rs-2>:31182,<domain-rs-3>:31183
+```
+
+### Conclusion
+At this point, you should be able to connect to the MongoDB deployment from outside the cluster. Make sure that you can resolve to the FQDNs for the replicaset members where you have the Mongo client installed.
```
Comment (irajdeep) [inline docs/external_access.md:89]: nit : maybe add an example here as to how to check if FQDN is resolvable using `dig` or something.

File: config/samples/external_access/cert-manager-certificate.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  name: cert-manager-certificate
+  namespace: <your-namespace>
```
Comment (bznein) [inline config/samples/external_access/cert-manager-certificate.yaml:6]: [nit] I had a quick look at other files in the `samples` dir and it looks like we generally do not specify an entry for `namespace` at all. To be consistent I'd say we should do the same in all files in this PR

File: config/samples/external_access/mongodb.com_v1_mongodbcommunity_cr.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
+        name: my-user-password
+      roles:
+        - name: clusterAdmin
+          db: admin
+        - name: userAdminAnyDatabase
+          db: admin
+      scramCredentialsSecretName: my-scram
+  additionalMongodConfig:
+    storage.wiredTiger.engineConfig.journalCompressor: zlib
```
Comment (bznein) [inline config/samples/external_access/mongodb.com_v1_mongodbcommunity_cr.yaml:35]: [nit] can we remove this? I think we left this once in a sample and we keep propagating it in every other sample even if not needed. We should keep it sample files as small as possible

File: docs/external_access.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+### Install cert-manager
+
+```sh
+kubectl create namespace cert-manager
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
+helm install \
+  cert-manager jetstack/cert-manager \
+  --namespace cert-manager \
+  --create-namespace \
```
Comment (bznein) [inline docs/external_access.md:14]: [nit] do we need this if we specified `kubectl create namespace cert-manager` as the first command?

File: docs/external_access.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+Apply the manifests.
+
+```sh
+kubectl apply -f config/samples/external_access/cert-manager-issuer.yaml
+kubectl apply -f config/samples/external_access/cert-manager-certificate.yaml
+```
+
+### Create the MongoDB deployment
+
+Edit ```config/samples/external_access/mongodb.com_v1_mongodbcommunity_cr.yaml```. Replace <mongodb-name> with the desired MongoDB deployment name -- this should be the same as in the previous step. Replace ```<domain-rs-1>```, ```<domain-rs-2>```, and ```<domain-rs-3>``` with the external FQDNs of the MongoDB replicaset members. Please remember that you should have the same number of entries in this section as the number of your replicaset members. You can also edit the ports for external access to your preferred numbers in this section -- you will remmeber to change them in the next step too. Change ```<your-admin-password>``` to your desired admin password for MongoDB.
```
Comment (bznein) [inline docs/external_access.md:52]:
```suggestion
Edit ```config/samples/external_access/mongodb.com_v1_mongodbcommunity_cr.yaml```. Replace <mongodb-name> with the desired MongoDB deployment name -- this should be the same as in the previous step. Replace ```<domain-rs-1>```, ```<domain-rs-2>```, and ```<domain-rs-3>``` with the external FQDNs of the MongoDB replicaset members. Please remember that you should have the same number of entries in this section as the number of your replicaset members. You can also edit the ports for external access to your preferred numbers in this section -- you will have to remember to change them in the next step too. Change ```<your-admin-password>``` to your desired admin password for MongoDB.
```

**Review by bznein — APPROVED** [review]
LGTM just a few minor comments!

File: docs/external_access.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+kubectl apply -f config/samples/external_access/external_services.yaml
+```
+
+### Retrieve the certificates from a MongoDB replicaset member
+
+```sh
+kubectl exec --namespace mcommunity -it <mongodb-name>-0 -c mongod -- bash
+```
+
+Once inside the container ```cat``` and copy the contents of ```/var/lib/tls/ca/ca.crt``` into a file on your local system. Also copy the contents of the ```.pem``` file in ```/var/lib/tls/server``` into a file on your local system.
```
Comment (chatton) [inline docs/external_access.md:78]: [nit]  We already have a CA creating from a file that is already on disk. The `.pem` file should be the only thing we need to access from the pod.

**Review by chatton — APPROVED** [review]
Looks great! This is going to be a massive help to a lot of users 🥇

File: docs/external_access.md (hunk 6 above)
Comment (priyolahiri) [reply docs/external_access.md]: Good point. Resolved in latest commit.

File: docs/external_access.md (hunk 4 above)
Comment (priyolahiri) [reply docs/external_access.md]: Good point. Resolved in latest commit.

File: config/samples/external_access/mongodb.com_v1_mongodbcommunity_cr.yaml (hunk 3 above)
Comment (priyolahiri) [reply config/samples/external_access/mongodb.com_v1_mongodbcommunity_cr.yaml]: Good point. Resolved in latest commit.

File: config/samples/external_access/cert-manager-certificate.yaml (hunk 2 above)
Comment (priyolahiri) [reply config/samples/external_access/cert-manager-certificate.yaml]: Good point. Resolved in latest commit.

File: docs/external_access.md (hunk 1 above)
Comment (priyolahiri) [reply docs/external_access.md]: Seems OOS to me.

**Pull request merged**